server/tserver/src/main/java/org/apache/accumulo/tserver/TabletServer.java

```diff
@@ -210,6 +215,8 @@ public PausedCompactionMetrics getPausedCompactionMetrics() {
   private final AtomicLong syncCounter = new AtomicLong(0);
 
   final OnlineTablets onlineTablets = new OnlineTablets();
+  private final Map<KeyExtent,AtomicLong> onDemandTabletAccessTimes =
```

Comment:
Wondering if it would simplify things if the last access time were stored in the tablet object.  Then this map does not need to be maintained and have the possibility to drift from the set of online tablets.   When this map is updated could instead call a method on the tablet.

Also wondering if it would be better for the tablet code to take ownership of updating the last access time rather than the tserver code.  Tserver code could read it and tablet code update it as needed.

Reply:
Opened #3263 for this